custom_conf.py

```diff
@@ -0,0 +1,244 @@
+
+    # Controls the existence of Previous / Next buttons at the bottom of pages
+    # Valid options: none, prev, next, both
+    'sequential_nav': "none",
+
+    # Controls if to display the contributors of a file or not
+    "display_contributors": True,
+
+    # Controls time frame for showing the contributors
+    "display_contributors_since": ""
+}
+
+# If your project is on documentation.ubuntu.com, specify the project
+# slug (for example, "lxd") here.
+slug = ""
+
+############################################################
+### Redirects
+############################################################
+
+# Set up redirects (https://documatt.gitlab.io/sphinx-reredirects/usage.html)
+# For example: 'explanation/old-name.html': '../how-to/prettify.html',
+# You can also configure redirects in the Read the Docs project dashboard
+# (see https://docs.readthedocs.io/en/stable/guides/redirects.html).
+# NOTE: If this variable is not defined, set to None, or the dictionary is empty,
+# the sphinx_reredirects extension will be disabled.
+redirects = {}
+
+############################################################
+### Link checker exceptions
+############################################################
+
+# Links to ignore when checking links
+linkcheck_ignore = [
+    'http://127.0.0.1:8000'
+    ]
+
+# Pages on which to ignore anchors
+# (This list will be appended to linkcheck_anchors_ignore_for_url)
+custom_linkcheck_anchors_ignore_for_url = []
+
+############################################################
+### Additions to default configuration
+############################################################
+
+## The following settings are appended to the default configuration.
+## Use them to extend the default functionality.
+
+# Remove this variable to disable the MyST parser extensions.
+custom_myst_extensions = []
+
+# Add custom Sphinx extensions as needed.
+# This array contains recommended extensions that should be used.
+# NOTE: The following extensions are handled automatically and do
+# not need to be added here: myst_parser, sphinx_copybutton, sphinx_design,
+# sphinx_reredirects, sphinxcontrib.jquery, sphinxext.opengraph
+custom_extensions = [
+    'sphinx_tabs.tabs',
+    'canonical.youtube-links',
+    'canonical.related-links',
+    'canonical.custom-rst-roles',
+    'canonical.terminal-output',
+    'notfound.extension',
+    'sphinx.ext.autodoc',
+    'sphinxcontrib.autodoc_pydantic',
+    ]
+
+# Add custom required Python modules that must be added to the
+# .sphinx/requirements.txt file.
+# NOTE: The following modules are handled automatically and do not need to be
+# added here: canonical-sphinx-extensions, furo, linkify-it-py, myst-parser,
+# pyspelling, sphinx, sphinx-autobuild, sphinx-copybutton, sphinx-design,
+# sphinx-notfound-page, sphinx-reredirects, sphinx-tabs, sphinxcontrib-jquery,
+# sphinxext-opengraph
+custom_required_modules = [
+    "autodoc_pydantic",
+    "pydantic-settings",
+    # package deps
```

Comment:
@k-dimple do you know if there is a better way to fix our autodocs generation other than include all our package deps here? ideally, it would just infer these through `pyproject.toml` but i didn't see any sphinx extensions to do that

Reply:
@rpocase I'll check and get back to you, however from a quick google search [sphinx-pyproject](https://pypi.org/project/sphinx-pyproject/) seems promising?

Reply:
thanks for looking! i dug into `sphinx-pyproject` and it doesn't seem to support our use case directly. it also requires you to run sphinx directly, not through the makefile, which breaks our workflows. what i'll likely end up doing is writing a small parser and embed that in custom_conf.py. i'll hold that for another MP though

Reply:
Some input from another TA: 
1. There is also the option of adding additional pre-build steps to .readthedocs.yaml and the Makefile. 
2. With a "new" starter pack that is coming soon, we will go back to the more straight-forward approach of using a requirements.txt file.

p.s. The "new" starter pack will move all our Canonical-related customisations into a separate module, so that we don't have to keep re-syncing our doc-set with changes (such as formatting & style) made to the starter pack.

Reply:
good to hear. seems like this will mostly resolve itself and we'll "fix" it later